Avoid initialization of graphical elements for a server.

RoosterDragon opened the pull request:
Avoid allocating memory and resources for graphical elements that will never be drawn when starting a dedicated server. This reduces the server memory footprint by approx 17 MiB.

pchote commented:
I finally summoned the effort to look at this properly (the glut of big prs recently has left me burnt out), and the changes look fine to me.  I haven't tested it, however.

Mailaender commented:
I think it is unfair to ask for huge refactorings when the "hack" is in line with the way the project is structured during pull requests provided by thirdpartys with clearly limited scope. A standalone true OpenRA dedicated server executable is of course a good idea. I filed https://github.com/OpenRA/OpenRA/issues/7252 so we don't forget.

Mailaender commented:
:+1: The changes look fine and they work in-game. To quickly test you can run `./launch-dedicated.sh` and `./launch-game.sh` to get two instances. Then direct connect to the server on the same machine.

RoosterDragon committed:
Avoid initialization of graphical elements for a server.

Avoid allocating memory and resources for graphical elements that will never be drawn when starting a dedicated server. This reduces the server memory footprint by approx 17 MiB.

pchote commented:
https://github.com/OpenRA/OpenRA/wiki/Changelog/_compare/5cc19e0663ffed25c0bf8158217e0e4370c59baa%5E...5cc19e0663ffed25c0bf8158217e0e4370c59baa
